language/stdlib/modules/LibraAccount.move

```diff
@@ -658,16 +658,22 @@ module LibraAccount {
         add_all_currencies: bool,
     ) {
         let new_account = create_signer(new_account_address);
-        Roles::new_child_vasp_role(parent, &new_account);
         VASP::publish_child_vasp_credential(
             parent,
             &new_account,
         );
+        Roles::new_child_vasp_role(parent, &new_account);
         Event::publish_generator(&new_account);
         add_currencies_for_account<Token>(&new_account, add_all_currencies);
         make_account(new_account, auth_key_prefix)
     }
 
+    spec fun create_child_vasp_account {
+
+        ensures Roles::spec_has_child_VASP_role_addr(new_account_address)
+                  ==> VASP::spec_is_vasp(new_account_address);
```

Comment:
Have you decided to change the direction of implication here and swap the lines above back, or keep it this way?